### ISSUE TITLE
dense_image_warp with other interpolation schemes

### ISSUE THREAD

**Issue by Luvideria** [tensorflow/addons#2733]
Currently, `dense_image_warp` uses only bilinear interpolation. It would be beneficial to use other interpolation schemes for better quality or different tradeoffs.

Other interpolation schemes are already available in other operations such as resize (Lanczos interpolation 3x3 or 5x5). 
`tf.image.resize(in, new_size, method = tf.image.ResizeMethod.LANCZOS5)`

**Relevant information** 
- Are you willing to contribute it (yes/no): yes, see next point
- I created a repo containing and explaining a working implementation of the *catmull-rom* (quite similar to Lanczos-2) interpolation as well as a comparison and a test
https://github.com/Luvideria/tensorflow-dense-warp-catmull-rom
- Are you willing to maintain it going forward? (yes/no): I don't know

**Which API type would this fall under (layer, metric, optimizer, etc.)**
image

**Who will benefit with this feature?**
Anyone using `dense_image_warp` and looking for more quality rather than speed.

**Comment by bhack** [comment]
To avoid duplication as we had a request also related to [elastic image warping in Keras-cv](https://github.com/keras-team/keras-cv/issues/163) can you try to open a ticket there?

**Comment by Luvideria** [comment]
You mean replying to the existing issue or raise a new issue?

**Comment by bhack** [comment]
As you like

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
